code/go/0chain.net/blobbercore/handler/storage_handler.go

```diff
@@ -325,56 +323,83 @@ func (fsh *StorageHandler) AddCollaborator(ctx context.Context, r *http.Request)
-	if r.Method == "GET" {
-		return nil, common.NewError("invalid_method", "Invalid method used. Use POST instead")
-	}
-	allocationTx := ctx.Value(constants.ALLOCATION_CONTEXT_KEY).(string)
+func (fsh *StorageHandler) GetFileStats(ctx context.Context, request *blobbergrpc.GetFileStatsRequest) (interface{}, error) {
+	//if r.Method == "GET" {
+	//	return nil, common.NewError("invalid_method", "Invalid method used. Use POST instead")
+	//}
+	//allocationTx := ctx.Value(constants.ALLOCATION_CONTEXT_KEY).(string)
+	allocationTx := request.Allocation
 	allocationObj, err := fsh.verifyAllocation(ctx, allocationTx, true)
 	if err != nil {
-		return nil, common.NewError("invalid_parameters", "Invalid allocation id passed."+err.Error())
+		return nil, errors.Wrap(err, "invalid allocation id in the request")
 	}
 	allocationID := allocationObj.ID
 
-	clientSign, _ := ctx.Value(constants.CLIENT_SIGNATURE_HEADER_KEY).(string)
+	clientSign := ctx.Value(constants.CLIENT_SIGNATURE_HEADER_KEY).(string)
+
+	// validate the signature from request
 	valid, err := verifySignatureFromRequest(allocationTx, clientSign, allocationObj.OwnerPublicKey)
-	if !valid || err != nil {
-		return nil, common.NewError("invalid_signature", "Invalid signature")
+	if err != nil || !valid {
+		return nil, errors.Wrap(err, "invalid signature passed in request")
 	}
 
 	clientID := ctx.Value(constants.CLIENT_CONTEXT_KEY).(string)
-	if len(clientID) == 0 || allocationObj.OwnerID != clientID {
-		return nil, common.NewError("invalid_operation", "Operation needs to be performed by the owner of the allocation")
+	if (clientID == "") || (allocationObj.OwnerID != clientID) {
+		return nil, errors.Wrap(errors.New("Invalid Operation"), "operation needs to be performed by allocation's owner")
 	}
 
-	_ = ctx.Value(constants.CLIENT_KEY_CONTEXT_KEY).(string)
-
-	pathHash, err := pathHashFromReq(r, allocationID)
-	if err != nil {
-		return nil, err
+	if request.PathHash == ""{
+		if request.Path == "" {
+			Logger.Error("Invalid request path passed in the request")
+			return nil, errors.Wrapf(errors.New("invalid request parameters"), "invalid request path")
+		}
+		request.PathHash = reference.GetReferenceLookup(allocationID, request.Path)
 	}
 
-	fileref, err := reference.GetReferenceFromLookupHash(ctx, allocationID, pathHash)
+	fileref, err := reference.GetReferenceFromLookupHash(ctx, allocationID, request.PathHash)
 
 	if err != nil {
-		return nil, common.NewError("invalid_parameters", "Invalid file path. "+err.Error())
+		Logger.Error("Invalid file path passed in the request")
+		return nil, errors.Wrapf(err, "invalid file path")
 	}
 
+	// additional check to ensure the FILE
 	if fileref.Type != reference.FILE {
-		return nil, common.NewError("invalid_parameters", "Path is not a file.")
+		Logger.Error("Path is not a file")
+		return nil, errors.Wrapf(errors.New("Invalid Parameters"), "path is not a file")
```

Comment:
same as `ErrInvalidOperator`. please move those errors to variables
```
var (
 ErrInvalidOperation = errors.New("Invalid Operation")
 ErrInvalidParameters = errorw.New("Invalid Parameters")
....
 

```

Reply:
Answered above.